src/prefect/server/models/deployments.py

```diff
@@ -230,6 +235,21 @@ async def update_deployment(
 
     requested_concurrency_limit_update = update_data.pop("concurrency_limit", "unset")
 
+    requested_global_concurrency_limit_update = update_data.pop(
+        "global_concurrency_limit_id", "unset"
+    )
+
+    if requested_global_concurrency_limit_update != "unset":
+        update_data["concurrency_limit_id"] = requested_global_concurrency_limit_update
+
+    if (
+        requested_global_concurrency_limit_update != "unset"
+        and requested_concurrency_limit_update != "unset"
+    ):
+        raise ValueError(
+            "Cannot provide both concurrency_limit and concurrency_limit_id"
+        )
```

Comment:
This validation should probably live on the `DeploymentUpdate` model as a [model validator](https://docs.pydantic.dev/latest/concepts/validators/#model-validators).